PubSub/src/Subscription.php

```diff
@@ -204,39 +204,30 @@ public function name()
-     *     @type string $pushConfig.oidcToken.audience Audience to be used when
-     *           generating OIDC token. The audience claim identifies the
-     *           recipients that the JWT is intended for. The audience value is
-     *           a single case-sensitive string. Having multiple values (array)
-     *           for the audience field is not supported. More info about the
-     *           OIDC JWT token audience here: https://tools.ietf.org/html/rfc7519#section-4.1.3
-     *           Note: if not specified, the Push endpoint URL will be used.
      *     @type int $ackDeadlineSeconds The maximum time after a subscriber
      *           receives a message before the subscriber should acknowledge the
      *           message.
-     *     @type bool $retainAckedMessages Indicates whether to retain
-     *           acknowledged messages.
-     *     @type Duration|string $messageRetentionDuration How long to retain
-     *           unacknowledged messages in the subscription's backlog, from the
-     *           moment a message is published. If `$retainAckedMessages` is
-     *           true, then this also configures the retention of acknowledged
-     *           messages, and thus configures how far back in time a `Seek`
-     *           can be done. Cannot be more than 7 days or less than 10 minutes.
-     *           If a string is provided, it should be as a duration in seconds
-     *           with up to nine fractional digits, terminated by 's', e.g
-     *           "3.5s". **Defaults to** 7 days.
+     *     @type array $deadLetterPolicy A policy that specifies the conditions
+     *           for dead lettering messages in this subscription. If
+     *           deadLetterPolicy is not set, dead lettering is disabled. The
+     *           Cloud Pub/Sub service account associated with this
+     *           subscriptions's parent project (i.e.,
+     *           service-{project_number}@gcp-sa-pubsub.iam.gserviceaccount.com)
+     *           must have permission to acknowledge messages on this
+     *           subscription.
+     *     @type string|Topic $deadLetterPolicy.deadLetterTopic The topic, or
+     *           name of the topic to which dead letter messages should be
+     *           published. Strings must be of format
+     *           `projects/{project}/topics/{topic}`. The Cloud Pub/Sub service
+     *           account associated with the enclosing subscription's parent
+     *           project (i.e., service-{project_number}@gcp-sa-pubsub.iam.gserviceaccount.com)
+     *           must have permission to publish to this topic. The operation
+     *           will fail if the topic does not exist. Users should ensure that
+     *           there is a subscription attached to this topic since messages
+     *           published to a topic with no subscriptions are lost.
+     *     @type int $deadLetterPolicy.maxDeliveryAttempts The maximum number of
```

Comment:
I don't see maxDeliveryAttempts field being used in any tests. We should write some tests with custom maxDeliveryAttempts and some with default maxDeliveryAttempts (5).

Reply:
added system test coverage.